### ISSUE TITLE
[Question] How to warm-start a SingleTaskGP

### ISSUE THREAD

**Issue by zhaojingyao** [pytorch/botorch#533]
# 🚀 Feature Request
In BO processs, I add a set of new data to SingleTaskGP with code `model = model.model.condition_on_observations(X=new_x,Y=new_y)` and re-train model with code `mll = ExactMarginalLogLikelihood(model.likelihood, model)
fit_gpytorch_model(mll)`

Is this the right way to warm-start in BO?

Thank you.

**Comment by Balandat** [comment]
By warm-starting you just mean re-starting the optimization of the hyperparmeters from the previous value? In theory `condition_on_observations` would do this, but the way this is implemented can in practice cause some issues with gradients: #356 (we're hoping to fix this at some point but haven't been able to dig into this enough so far). For now please use `condition_on_observations` only if you don't re-fit the model (typically for generating "fantasy models"). 

Instead, you should instantiate a new model and loading the state dict (this is the pytorchic way) like so

```
old_state = old_model.state_dict()

new_model = SingleTaskGP(full_x, full_y)
new_model.load_state_dict(old_state)
mll = ExactMarginalLogLikelihood(model.likelihood, model)
fit_gpytorch_model(mll)
```

**Comment by Balandat** [comment]
Assuming this solved your issue, feel free to re-open if not.
